### ISSUE TITLE
Can't change bowls planks cost

### ISSUE THREAD

**Issue by Insane96** [MehVahdJukaar/sawmill#25]
MC 1.20.1
sawmill-1.20-1.3.4.jar
Forge 47.2.20

![image](https://github.com/MehVahdJukaar/sawmill/assets/7185104/a442aa75-fe58-4375-99b6-363572859c62)

**Comment by MehVahdJukaar** [comment]
aparently its broken for any item that does not belong to a wood set (except for stick that was hardcoded). I'll figure out a better solution

**Comment by Insane96** [comment]
Thanks for the update, but now there's some kind of rounding problem
![image](https://github.com/MehVahdJukaar/sawmill/assets/7185104/e7daec6e-e17d-47d8-8ee5-3cb63ad538e8)

**Comment by MehVahdJukaar** [comment]
dang it 💀

**Comment by MehVahdJukaar** [comment]
so whats going on i believe is that my system is still checking normal recipes for that. special costs are added early on, then it iterates over all recipes and always keeps the lowest price it finds. this means that configs are only really meant to add discounts to items and cant make them more expensive like in this case

**Comment by Insane96** [comment]
Then why the default recipe gives 7 bowls? 3 planks for 4 bowls should be 4 planks for 5.3 bowls, and not 7, at most 6.

**Comment by MehVahdJukaar** [comment]
yeah its because of the discount logic. that stuff is pretty arbitrary as i tuned to to make sure certain vanilla recipes would get some discount but not too much. discount is proportional to items crafted as a percentage and only applies when certain conditions on the output original count are valid. I tweaked it a bit so it will give less discount overall as it now takes the average of old and dicounted price. the latter can also be changed via configs
